src/http/cookie.cr

```diff
@@ -0,0 +1,83 @@
+require "cgi"
+require "./common"
+
+module HTTP
+  class Cookie
+    def self.parse(str : String) : Cookie
+      # TODO: error handling, empty string + no name=value pair
+      parts = str.split(/[;]\s?/)
+      name, value = parts[0].split(/\s*=\s*/, 2)
+
+      cookie = HTTP::Cookie.new(name, value)
+      (1...parts.size).each do |i|
+        part = parts[i]
+        case part
+        when .=~ /^\s*path=/i
+          match = part.match(/^\s*path\=(.+)/i)
+          cookie.path = match[1] if match
+        when .=~ /^\s*domain=/i
+          match = part.match(/^\s*domain\=(.+)/i)
+          cookie.domain = match[1] if match
```

Comment:
Reusing the last match should be possible here, `$~[1]` or `Regex.last_match[1]`